Tests/SwiftCuckooTests/LapTests.swift

```diff
@@ -0,0 +1,133 @@
+import Foundation
+import Testing
+@testable import SwiftCuckoo
+
+private extension Lap.Identifier {
+    static let lapTest = Lap.Identifier(rawValue: "lapTest")
+}
+
+final class LapTests {
+
+    /// Tests that starting a lap sets the start time
+    @Test("Start lap should start time")
+    func testStartLap() throws {
+        // Arrange: Create a lap object with a defined start time
+        let startTime: Date = .distantPast
+        var lap = Lap(id: .lapTest, startTime: startTime)
+
+        // Act: Start the lap
+        try lap.start(on: startTime)
+
+        // Assert: Verify that lap time is set
+        #expect(
+            lap.startTime == startTime,
+            "The lap's start time should be set to provided start time")
+    }
+
+    /// Test that starting a lap twice will throw an error
+    @Test("Starting a lap twice should throw an error")
+    func testStartingLapTwiceThrowsError() throws {
+
+        // Arrange: Create new lap
+        var lap = Lap(id: .lapTest, startTime: .now)
+        try lap.start()
+
+        // Act & Assert: Attempt to start the lap again and expect an error
+        #expect(
+            throws: Lap.Error.lapActive,
+            "Starting a lap that is already active should throw an error",
+            performing: {
+                try lap.start()
+            }
+        )
+    }
```

Comment:
I am 90% sure that this test could fail since our CI is not working properly, so I will fix it soon.
Have you run it locally?

Reply:
@Tavernari Not sure if my xcode is configured to run Swift 6 testing